### ISSUE TITLE
IDE0001 false positive on nullable attributes

### ISSUE THREAD

**Issue by RenderMichael** [dotnet/roslyn#75570]
**Version Used**: 

Visual Studio 17.12.0 Preview 3.0

**Steps to Reproduce**:

Perhaps this is a bug in nullability analysis, where the nullability of the generic arguments should more opportunistically recognize `NotNullIfNotNullAttribute` paired with the knowledge that the generic argument in `IEnumerable<C1>` is not null.

```csharp
static C2[] ConvertAll(IEnumerable<C1> values)
{
    return values.Select<C1, C2>(Convert).ToArray(); // IDE0001 on <C1, C2>
}

class C1;

class C2;

[return: NotNullIfNotNull(nameof(value))]
private static C2? Convert(C1? value) => value is null ? null : new C2();
```
**Diagnostic Id**:

`IDE0001: Name can be simplified`

**Expected Behavior**:

IDE0001 should not fire (or the fixed code should not have nullability warnings, as mentioned above). 

**Actual Behavior**:

Suggests a fixer that changes the code to this:

```csharp
static C2[] ConvertAll(IEnumerable<C1> values)
{
    return values.Select(Convert).ToArray(); // CS8619: Nullability of reference types does not match
}

class C1;

class C2;

[return: NotNullIfNotNull(nameof(value))]
private static C2? Convert(C1? value) => value is null ? null : new C2();
```

**Comment by RenderMichael** [comment]
@CyrusNajmabadi With this issue closed, is the guidance now to include the generic parameters in the exemplary `Select` call, or to exclude them?

**Comment by CyrusNajmabadi** [comment]
You need the generic type parameters as the inferred ones will cause a warning.  But the fixer is fixed so that it will now no longer remove these.
